spec.html

```diff
@@ -40513,13 +40511,9 @@ <h1>Promise.prototype.then ( _onFulfilled_, _onRejected_ )</h1>
 
         <emu-clause id="sec-performpromisethen" aoid="PerformPromiseThen">
           <h1>PerformPromiseThen ( _promise_, _onFulfilled_, _onRejected_ [ , _resultCapability_ ] )</h1>
-          <p>The abstract operation PerformPromiseThen takes arguments _promise_, _onFulfilled_, and _onRejected_ and optional argument _resultCapability_. It performs the &ldquo;then&rdquo; operation on _promise_ using _onFulfilled_ and _onRejected_ as its settlement actions. If _resultCapability_ is passed, the result is stored by updating _resultCapability_'s promise. If it is not passed, then PerformPromiseThen is being called by a specification-internal operation where the result does not matter. It performs the following steps when called:</p>
+          <p>The abstract operation PerformPromiseThen takes arguments _promise_, _onFulfilled_, and _onRejected_ and optional argument _resultCapability_ (a PromiseCapability Record). It performs the &ldquo;then&rdquo; operation on _promise_ using _onFulfilled_ and _onRejected_ as its settlement actions. If _resultCapability_ is passed, the result is stored by updating _resultCapability_'s promise. If it is not passed, then PerformPromiseThen is being called by a specification-internal operation where the result does not matter. It performs the following steps when called:</p>
           <emu-alg>
             1. Assert: IsPromise(_promise_) is *true*.
-            1. If _resultCapability_ is present, then
-              1. Assert: _resultCapability_ is a PromiseCapability Record.
-            1. Else,
-              1. Set _resultCapability_ to *undefined*.
```

Comment:
I'm guessing you removed this thinking that a not-present parameter would be `*undefined*` anyway. However, that's not a thing for abstract operations.

Rather than reinstate these lines, you could convert `_resultCapability_` to a required parameter, and pass `*undefined*` at the 2 calls where it's currently omitted. (I seem to recall some editor (@syg?) expressing approval for changing optional AO parameters to required.)

Reply:
hmm, i'm not a fan of that idea, so i'll partially revert this back.